### DISCUSSION TITLE
How can i use a custom font localy?

### DISCUSSION THREAD

**Discussion by Ahmad-Fathy** [filamentphp/filament#6869] [General]
When i use a custom font i but its link in `config/filament.php` (google_fonts)" and change the font name in `tailwind.config.js`
but how can i use the custom font localy?

**Comment by danharrin** [comment]
You can add it to your public directory and then load it in the CSS file of your custom theme?

**Reply by Ahmad-Fathy** [reply]
I just solved it .. thank you

**Reply by Hegabovic** [reply]
do you mind giving a detailed steps in order to use this font family (documentation is not helping much) as i want to use [Cairo Font Family](https://fonts.google.com/specimen/Cairo)  in my Arabic dashboard

**Reply by danharrin** [reply]
In v3: https://filamentphp.com/docs/3.x/panels/themes#changing-the-font-provider

**Reply by wq9578** [reply]
Changing the font provider for panels works, but how about for the login page (or the other Auth pages in namespace `Filament\Pages\Auth`, or custom pages in namespace `App\Filament\Pages`?
It would be desirable to use the same font provider across the whole app (maybe even setting it globally).

**Reply by wq9578** [reply]
Solved. The default is set in class AdminPanelProvider.

**Reply by Hegabovic** [reply]
the question was meant for version 2 but in version 3 you can add i to the panel provider
